Mlem/App/Views/Shared/Search/SearchView.swift

```diff
@@ -0,0 +1,155 @@
+    @State var page: Page = .home
+
+    @State var selectedTab: Tab = .communities
+    @State var resultsScrollToTopTrigger: Bool = false
+    
+    @State var communities: [Community2] = []
+    @State var people: [Person2] = []
+    @State var instances: [InstanceSummary] = []
+    
+    var body: some View {
+        content
+            .navigationTitle("Search")
+            .navigationBarTitleDisplayMode(.large)
+            .navigationSearchBar {
+                SearchBar("Communities, Users & Instances", text: $query, isEditing: $isSearching)
+                    .showsCancelButton(page != .home)
+                    .onCancel {
+                        page = .home
+                        if !query.isEmpty {
+                            query = ""
+                            Task { await refresh() }
+                        }
+                        resultsScrollToTopTrigger.toggle()
+                    }
+                    .focused($searchBarFocused)
+            }
+            .navigationSearchBarHiddenWhenScrolling(false)
+            .toolbar { PasteLinkButtonView() }
+            .onChange(of: query) {
+                if page != .home {
+                    if query.isEmpty {
+                        page = .recents
+                    } else {
+                        page = .results
+                    }
+                }
+            }
+            .onChange(of: isSearching) {
+                if isSearching, query.isEmpty {
+                    page = .recents
+                }
+            }
+            .task(id: query, priority: .userInitiated) { @MainActor in
+                guard !hasAppeared || searchBarFocused else { return }
+                hasAppeared = true
+                await refresh()
+            }
+            .onChange(of: appState.firstApi.actorId) {
+                Task { @MainActor in
+                    await refresh()
+                }
+            }
+            .onChange(of: selectedTab) {
+                Task { @MainActor in
+                    do {
+                        switch selectedTab {
+                        case .communities:
+                            if communities.isEmpty {
+                                communities = try await appState.firstApi.searchCommunities(query: query, page: 1, limit: 20)
+                            }
+                        case .users:
+                            if people.isEmpty {
+                                people = try await appState.firstApi.searchPeople(query: query, page: 1, limit: 20)
+                            }
+                        case .instances:
+                            if instances.isEmpty {
+                                instances = try await MlemStats.main.searchInstances(query: query)
+                            }
+                        }
+                    } catch {
+                        handleError(error)
+                    }
+                }
```

Comment:
Ditto comments above re: main actor work. Similar issue noted below--probably worth defining a little

```
@MainActor
func setCommunities(_ newCommunities: [Community2]) {
  communities = newCommunities
}
```

etc. for people, instances, so that this can just be

`setCommunities(try await appState.firstApi.searchCommunities(query: query, page: 1, limit: 20))`

Reply:
I've marked the properties themselves as `@MainActor`, which I think is the same as doing this? XCode doesn't seem to be complaining about it.